grove/pyquil_utilities.py

```diff
@@ -0,0 +1,92 @@
+"""Module containing utilities that will eventually be implemented in pyquil."""
+
+from pyquil.quilbase import (ACTION_INSTANTIATE_QUBIT,
+                             ACTION_INSTALL_INSTRUCTION,
+                             ACTION_RELEASE_QUBIT)
+from pyquil.quilbase import Gate, Qubit
+from pyquil.resource_manager import merge_resource_managers, ResourceManager
+
+
+def prog_equality(prog1, prog2):
+    """Tests to see if two programs are equal based on their out methods
+
+    :param prog1: The first Program to compare
+    :param prog2: The second program to compare
+    """
+    return prog1.out() == prog2.out()
+
+
+def non_action_insts(prog):
+    """Returns quantum gates, and classical control flow of the program, ignoring qubit allocation,
+     and qubit release.
+
+    :param prog: The program to return instructions for.
+    :return: The length of the program, ignoring the irrelevant actions.
+    :rtype: list
+    """
+    insts = []
+    for action in prog:
+        if action[0] not in (ACTION_INSTANTIATE_QUBIT, ACTION_RELEASE_QUBIT):
+            insts.append(action)
+    return insts
+
+
+def prog_len(prog):
+    """Returns the length of of a program, defined as the number of instructions.
+
+    :return: The length of prog.
+    :rtype: int
+    """
+    length = 0
+    for action in prog:
+        if action[0] == ACTION_INSTANTIATE_QUBIT:
+            continue
+        length += 1
+    return length
+
+
+def synthesize_programs(*progs):
```

Comment:
After reading this, I went back to look where you actually use this. It only appears in tests, so we can actaully scratch this beast. Please remove!

Reply:
@jotterbach I wrote this so I could make sure `Program`'s being produced were what I expected them to be. Currently we don't have a good way to test for `Program` equality.

Reply:
but it's not used anywhere. So we might as well remove it. Don't introduce code that is not used or out-of-scope.